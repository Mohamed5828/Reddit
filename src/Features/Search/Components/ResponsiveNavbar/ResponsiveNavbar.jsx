import { useState, useEffect } from "react";
import Dropdown from "react-bootstrap/Dropdown";
import { NavLink } from "react-router-dom";
import { StyledDropdown } from "./ResponsiveNavbar.styled";
/**
 * Component that  Main Links component but in responsive form.
 *
 * @Component
 * @returns {React.Component}
 */
const ResponsiveNavbar = () => {
  const pathname = window.location.pathname;
  const firstTwo = pathname.substring(1);
  const secondTwo = firstTwo.charAt(7).toUpperCase() + firstTwo.slice(8);
  const [Title, setTitle] = useState(secondTwo);
  // const [Title, setTitle] = useState(props.act);

  useEffect(() => {
    // console.log(props.act);
    const pathname = window.location.pathname;
    console.log(pathname);
    const firstTwo = pathname.substring(1);
    const secondTwo = firstTwo.charAt(7).toUpperCase() + firstTwo.slice(8);
    setTitle(secondTwo);
    return () => {};
  }, [pathname]);
  return (
    <StyledDropdown
      id="dropdown-basic-button"
      title={Title}
      variant={"transparent"}
      size="md"
    >
      <Dropdown.Item
        as={NavLink}
        to="/search/posts"
        onClick={() => {
          setTitle("Posts");
        }}
      >
        Posts
      </Dropdown.Item>
      <Dropdown.Item
        as={NavLink}
        to="/search/comments"
        onClick={() => {
          setTitle("Comments");
        }}
      >
        Comments
      </Dropdown.Item>
      <Dropdown.Item
        as={NavLink}
        to="/search/communities"
        onClick={() => {
          setTitle("Communities");
        }}
      >
        Communities
      </Dropdown.Item>
      <Dropdown.Item
        as={NavLink}
        to="/search/people"
        onClick={() => {
          setTitle("People");
        }}
      >
        People
      </Dropdown.Item>
    </StyledDropdown>
  );
};

export default ResponsiveNavbar;
