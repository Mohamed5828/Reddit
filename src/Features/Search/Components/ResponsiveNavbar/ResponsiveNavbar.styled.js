import styled from "styled-components";
// import Dropdown from "react-bootstrap/Dropdown";
import DropdownButton from "react-bootstrap/DropdownButton";
export const StyledDropdown = styled(DropdownButton)`
  display: none;
  transition: 0.3s;
  @media (max-width: 635px) {
    display: block;
  }
  & button {
    /* background-color: transparent; */
    background-color: ${({ theme }) => theme.background.active};

    padding: 12px 20px;
    font-weight: 700;
    color: ${({ theme }) => theme.color.titleText_of_link};
    /* color: #1a1a1b; */
    display: flex;
    align-items: center;
    justify-content: center;
    text-align: center;
    position: relative;
    border-radius: 9999px;
    width: auto;
    border: none;
    outline: none;
    cursor: pointer;
    letter-spacing: unset;
    line-height: 17px;
    text-transform: unset;
    min-height: 32px;
    min-width: 32px;
  }
  & button:hover {
    background-color: ${({ theme }) => theme.background.fill};
    color: ${({ theme }) => theme.color.titleText_of_link};
  }
  .dropdown-item.active {
    background-color: ${({ theme }) => theme.background.muted};
    color: ${({ theme }) => theme.color.titleText_of_link};
  }
  .dropdown-menu.show {
    background-color: ${({ theme }) => theme.background.active};
  }
  .dropdown-item {
    color: ${({ theme }) => theme.color.titleText_of_link};
  }
  .dropdown-item:hover {
    background-color: ${({ theme }) => theme.background.muted};
  }
`;
