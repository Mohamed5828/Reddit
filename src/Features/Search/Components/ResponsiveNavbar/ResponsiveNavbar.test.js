import { shallow } from "enzyme";
import ResponsiveNavbar from "./ResponsiveNavbar";
describe("Responsive Navbar", () => {
  it("this is a test for Responsive Navbar", () => {
    expect(shallow(<ResponsiveNavbar />)).toMatchSnapshot();
  });
});
