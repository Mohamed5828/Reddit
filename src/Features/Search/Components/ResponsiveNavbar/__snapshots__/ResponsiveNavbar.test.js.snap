// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsive Navbar this is a test for Responsive Navbar 1`] = `
<Styled(DropdownButton)
  id="dropdown-basic-button"
  size="md"
  title=""
  variant="transparent"
>
  <DropdownItem
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    onClick={[Function]}
    to="/search/posts"
  >
    Posts
  </DropdownItem>
  <DropdownItem
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    onClick={[Function]}
    to="/search/comments"
  >
    Comments
  </DropdownItem>
  <DropdownItem
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    onClick={[Function]}
    to="/search/communities"
  >
    Communities
  </DropdownItem>
  <DropdownItem
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    onClick={[Function]}
    to="/search/people"
  >
    People
  </DropdownItem>
</Styled(DropdownButton)>
`;
